src/parents.js

```diff
@@ -35,8 +35,28 @@ export function set (node, parent, { force } = {}) {
 		// We assume that if the node already has a parent, its subtree will also have parents
 		return false;
 	}
-	const metadata = findParentProp(node, parent);
-	parentMap.set(node, {parent, ...metadata});
+
+	let details = {};
+	const parentProps = properties[parent?.type] ?? [];
+
+	// Find the property name and index of the child node in the parent
+	for (const property of parentProps) {
+		const child = parent[property];
+		// If the child is an array, get the property and index
+		if (Array.isArray(child)) {
```

Comment:
(Non-blocking suggestion)

```suggestion
		if (Array.isArray(child)) {
			// When the child is an array, we also need an index to get from parent to child
```
or just
```suggestion
		if (Array.isArray(child)) {
			// When the child is an array, we also need to figure out index
```

`if(Array.isArray(child))` is pretty self-explanatory, so commenting "If the child is an array" duplicates info. The important bit here is that if the child is an array the property alone is not enough.